test/result_tests.cpp

```diff
@@ -267,6 +267,19 @@ TEST(ResultTests, StringToViewNoThrow) {
   }
 }
 
+TEST(ResultTests, MaybeError) {
+  // GIVEN two result<T> values
+  const Result<double> a = 6.5;
+  const Result<int> b = make_unexpected(Unknown());
```

Comment:
```suggestion
  const Result<int> b = make_unexpected(Unknown{});
```

Reply:
Unknown is a constexpr lambda that returns an object so it has to be called this way... one thing I've been wondering about is should I put this all in some namespace?